protocols/floodsub/src/layer.rs

```diff
@@ -206,7 +206,7 @@ impl Floodsub {
 
         let self_subscribed = self.subscribed_topics.iter().any(|t| message.topics.iter().any(|u| t == u));
         if self_subscribed {
-            self.received.add(&message);
+            self.received.add(&message).expect("Not enough space to store this item, rebucketing failed.");
```

Comment:
I am not sure panicking is the right thing to do here. We should probably handle this more gracefully. Do you think this happening is so unlikely?

> When the filter returns NotEnoughSpace, the element given is actually added to the filter, but some random other element gets removed. This could be improved by implementing a single-item eviction cache for that removed item.

Reply:
Actually, the cuckoofilter v0.3.2 will panic directly in the `add` function when map is full.

https://github.com/axiomhq/rust-cuckoofilter/blob/v0.3.2/src/lib.rs#L158